prismic-model/js/parts/promo.js

```diff
@@ -1,42 +1,28 @@
 // @flow
+import image from './image';
+import text from './text';
+
 export default {
-  'type': 'Slices',
-  'config': {
-    'label': 'Promo',
-    'choices': {
-      'editorialImage': {
-        'type': 'Slice',
-        'fieldset': 'Editorial image',
-        'config': {
-          'label': 'Editorial image'
+  type: 'Slices',
+  config: {
+    label: 'Promo',
+    choices: {
+      editorialImage: {
+        type: 'Slice',
+        fieldset: 'Editorial image',
+        config: {
+          label: 'Editorial image'
         },
         'non-repeat': {
-          'caption': {
-            'type': 'StructuredText',
-            'config': {
-              'label': 'Promo text',
-              'single': 'paragraph'
+          caption: {
+            type: 'StructuredText',
+            config: {
+              label: 'Promo text',
+              single: 'paragraph'
             }
           },
-          'image': {
-            'type': 'Image',
-            'config': {
-              'label': 'Promo image',
-              'thumbnails': [ {
-                'name': '32:15',
-                'width': 3200,
-                'height': 1500
-              }, {
-                'name': '16:9',
-                'width': 3200,
-                'height': 1800
-              }, {
-                'name': 'square',
-                'width': 3200,
-                'height': 3200
-              } ]
-            }
-          }
+          image: image('Promo image'),
+          link: text('Link override')
```

Comment:
I was going to make this a link, but you can't do relatie links as it forces it to `https://relative/path`.